src/lib/updating-element.ts

```diff
@@ -90,9 +99,33 @@ type AttributeMap = Map<string, PropertyKey>;
 
 export type PropertyValues = Map<PropertyKey, unknown>;
 
-// serializer/deserializers for boolean attribute
-const fromBooleanAttribute = (value: string) => value !== null;
-const toBooleanAttribute = (value: string) => value ? '' : null;
+export const defaultConverter: ComplexAttributeConverter = {
+
+  toAttribute(value: any, type?: any) {
+    switch (type) {
+    case Boolean:
+      return value ? '' : null;
+    case Object:
+    case Array:
+      return JSON.stringify(value);
```

Comment:
Based on experience with this being a footgun for users, consider anything better?

Reply:
Added this #375 to ensure this is well documented.